packages/emotion/src/createCssFunction.ts

```diff
@@ -0,0 +1,64 @@
+import { css as emCss, SerializedStyles, Theme } from '@emotion/react'
+import { CSSInterpolation } from '@emotion/serialize'
+import { StyleGenerator } from '@xstyled/system'
+import { createTransform } from '@xstyled/core'
+
+const styleToString = (style: any, props: any): any => {
+  if (Array.isArray(style))
+    return style.reduce((str, style) => str + styleToString(style, props), '')
+  if (typeof style === 'function') return styleToString(style(props), props)
+  return style
+}
+
+interface Props {
+  theme?: Theme
+}
+
+interface CSSInterpolationFn {
+  (props: Props): CSSInterpolation
+}
+
+export interface SerializedStylesFn {
+  (props: Props): SerializedStyles
+}
+
+export interface XCSSFunction {
+  (fn: CSSInterpolationFn): SerializedStylesFn
+  (...args: CSSInterpolation[]): SerializedStylesFn
+  (
+    strings: TemplateStringsArray,
+    ...rawArgs: CSSInterpolation[]
+  ): SerializedStylesFn
+  (
+    strings: TemplateStringsArray | CSSInterpolation | CSSInterpolationFn,
+    ...rawArgs: CSSInterpolation[]
+  ): SerializedStylesFn
+}
+
+export const createCssFunction = <TGen extends StyleGenerator>(
```

Comment:
createCssFunction is called numerous times from createCss with the same args. Should we memoize it? If so, maybe we should use something like reselect's [defaultMemoize](https://github.com/reduxjs/reselect/blob/f4a21736ecfb8ac80f44799647ce81bb158a671f/src/index.js#L21-L34) because it only remembers the most recent, so there's no memory leak.

Reply:
(I'm thinking of memoization here to reduce memory consumption just as much as performance, since there's no need to have multiple constructed identical css functions)

Reply:
It costs literally nothing, in fact it is just a binding that returns another function. I think there is no need to memoize it except if we see a problem in performance in real application.

Reply:
Oh, you're completely right. I was thinking of `styled` with all the per-element functions hanging off it :flushed: